### PR TITLE
meta-wpe: remove hard-coded RPI configuration and obsolete rpi-config recipe

### PR DESCRIPTION
**Pull request by hhromic**
The layer is currently hard-coding the following RPI-specific configurations in `layer.conf` and the `rpi-config` recipe:

* HDMI output mode (rpi-config recipe)
* splash/overscan disabling (rpi-config recipe)
* boot delay (rpi-config recipe)
* overclocking (layer.conf)
* memory split (layer.conf)

If building for other machines than RPI, these variables/customisations might be confusing or inappropriate.

Instead, the user should be instructed, e.g. in the Wiki, to set these variables in `local.conf` according to his/her own needs and particular machine.
For example, the Raspberry Pi 3 is down-graded by the forced overclocking settings because these are for a Raspberry Pi 2 which runs slower.

Upstream `meta-raspberrypi` is now supporting the customisations done by the `rpi-config` recipe here, rendering it obsolete so it was removed.

If merged, I will update the Wiki to instruct users about these RPI-specific customisations.

### CODE REVIEW

**Comment by hhromic** [comment]
This is the proposed addition to the Wiki about `local.conf` customisation:

```
# Set HDMI video mode
# for 720p use mode 4 (optimum) and for 1080p use mode 16
HDMI_GROUP = "1"
HDMI_MODE = "4"

# Disable rainbow splash, overscan and boot delay
DISABLE_SPLASH = "1"
DISABLE_OVERSCAN = "1"
BOOT_DELAY = "0"

# Enable Turbo mode on Rpi2
ARM_FREQ = "1000"
CORE_FREQ = "500"
SDRAM_FREQ = "500"
OVER_VOLTAGE = "6"

# Additional overclocking of GPU components
# (core_freq, h264_freq, isp_freq and v3d_freq)
GPU_FREQ = "500"

# Set GPU/CPU memory split
GPU_MEM_256 = "128"
GPU_MEM_512 = "196"
GPU_MEM_1024 = "396"
```

**Comment by kraj** [comment]
Please add the above to README.md and send a v2

**Comment by hhromic** [comment]
Updated :)

**Comment by kraj** [comment]
yes thats fine too.

**Comment by kraj** [comment]
Generally, having them in code path results in better out of box experience, if we want to make it configurable we should tell the user in very precise terms what to do before launching the builds.
otherwise, I am expecting questions on performance and all leading to this

**Comment by hhromic** [comment]
Hi guys, thanks for your comments.

@wouterlucas, I also think the README should crisp and instructive on how to start in *general* and then each specific board have their details independently in a separate documentation. Perhaps the Wiki should be cleaner and better organised. Or perhaps we can adopt read-the-docs as `meta-raspberrypi` does. I can volunteer myself to tidy this up if you need.

Regarding the out-of-the box experience (@kraj's concerns), I believe this layer should follow the "Yocto ways", and I believe `meta-wpe` is a **middleware layer** for any final product based on WPE. This means that this layer should be used as a dependency to an end-user's **final application layer** instead of directly. I know this introduces a bit more hassle but is good engineering practice and we should promote it.

For example, I'm building a web-kiosk with WPE for the RPI0W. For this I created a `meta-wpe-kiosk` application layer that in turn depends on `meta-wpe` as middleware to provide WPE. Then, in my own layer, I put the customisations I need for my application to run on the RPI0W and my display such as overclocking, Wifi settings or resolution. As a middleware, `meta-wpe` shouldn't enforce any of these things.

To summarise, I can work on improving the documentation in the Wiki to effectively instruct users (specially novice ones like me) on how to use this layer properly and with good practices. If you agree with this, I will update the PR again to tidy up the README and improve the Wiki.

**Comment by hhromic** [comment]
Btw, in case you wonder, I'm not making any sort of commercial product with the web-kiosk, is just for learning and deploying in a non-profit computer museum for interactive displays :)

**Comment by hhromic** [comment]
I updated the PR now with my proposed changes for the README.
In case you accept it, I will further work out the Wiki for clarity and documentation.

**Comment by kraj** [comment]
Think of this as an opportunity to improve out of box experience for someone who is not yocto/oe hacker. I think more the customization to be done manually more difficult and error prone it becomes.

**Comment by hhromic** [comment]
I think good documentation and clean code are powerful resources. Specially for maintenance of the layer. IMO, we can provide both, clean out-of-the-box and clean yocto hacker experiences with clear up-to-date instructions/documentation and correct code in the layer.
As someone that started with this layer and yocto completely from scratch in less than a year, and provided patches precisely to improve the out-of-the-box experience, I can document in the Wiki how to get started easily and clearly for the newcomers. I've been building a gist with my personal workflow since a while and I think I can use it as a base: https://gist.github.com/hhromic/9de455e943a52e0a641cf5171cc18e86
What do you think?

**Comment by hhromic** [comment]
Hi @wouterlucas , thanks for the feedback, I'm happy to help.
Yes, my gist is primarily a recipe for myself to build meta-wpe for my RPI0W, however is a bit outdated (for example I don't use rocko anymore), and I believe the only armv6-dependant aspect is setting MACHINE to raspberrypi0-wifi.

I would like to keep helping too on cleaning/tidying up `meta-wpe`.
I got recently confused because I came up to the existence of this repo/layer: [meta-webkit`(https://github.com/Igalia/meta-webkit). What is the relationship between that layer and `meta-wpe` ?  looks like duplicated efforts?

Nice to see those new features you mention too. I will take a look to learn how they work and help documenting when I can.

**Comment by wouterlucas** [comment]
There is definitely a level of duplication between meta-wpe and meta-webkit. Main difference is meta-webkit is focussed on building directly from trunk to deal with upstream. Limited to a lot less targets (mainly RPI and IMX at this stage) while meta-wpe focusses on the downstream port including support for a lot more devices. We tend to integrate with a lot of RDK-V devices (STB oriented stacks) and thus provide a little more focus on multimedia support (e.g. YT cert). Patches that are maintained over here may be a little more crude, though needed, but may never go upstream (e.g. Gstreamer and Graphics related fixes) to meet certain certifications.

I'm planning to syncup with meta-webkit's efforts on the include files and seperation, @clopez efforts have been awesome. But for now we've agreed to keep them separate. 

And yeah help is always welcome :-)
